controllers/keystoneapi_controller.go

```diff
@@ -437,12 +450,11 @@ func (r *KeystoneAPIReconciler) reconcileUpgrade(ctx context.Context, instance *
 func (r *KeystoneAPIReconciler) reconcileNormal(ctx context.Context, instance *keystonev1.KeystoneAPI, helper *helper.Helper) (ctrl.Result, error) {
 	r.Log.Info("Reconciling Service")
 
-	if !controllerutil.ContainsFinalizer(instance, helper.GetFinalizer()) {
-		// If the service object doesn't have our finalizer, add it.
-		controllerutil.AddFinalizer(instance, helper.GetFinalizer())
-		// Register the finalizer immediately to avoid orphaning resources on delete
-		err := r.Update(ctx, instance)
-
+	// If the service object doesn't have our finalizer, add it.
+	controllerutil.AddFinalizer(instance, helper.GetFinalizer())
+	r.Log.Info("Adding Finalizer")
+	// Register the finalizer immediately to avoid orphaning resources on delete
+	if err := r.Update(ctx, instance); err != nil {
```

Comment:
I don't think we want to change this. we only want to add the finalizer and update the object if it is not present.

Reply:
right I'm changing it back